### ISSUE TITLE
How can I use Fabric attributes in a LightningModule?

### ISSUE THREAD

**Issue by EricKani** [Lightning-AI/lightning#17358]
### Description & Motivation

For example, I want to use fabric.world_size in LightningModule. How should I do... Should I pass the instance as a parameter to LightningModule?

cc @carmocca @justusschock @awaelchli @borda

**Comment by awaelchli** [comment]
Hey @EricKani 

As soon as you pass the LM to `fabric.setup()`, it will connect the fabric instance the LM and you can call `self.fabric` inside the LightningModule to access its properties.

**Comment by awaelchli** [comment]
I think we should document it on this page: https://lightning.ai/docs/fabric/stable/guide/lightning_module.html

**Comment by carmocca** [comment]
`trainer.world_size` will also route it to fabric.

**Comment by EricKani** [comment]
> 

Thanks! This works!

I found a new issue when I use lightning-fabric framework. In Lightning1.9.3 version, there seems no fabric.clip_gradients() or related implementation.  I implement gradient clip like below, is this ok~?
`
fabric.backward(loss)
torch.nn.utils.clip_grad_norm_(model.parameters(), 2.0)
optimizer.step()`

Best,
Eric

**Comment by awaelchli** [comment]
Sure, it will work for most strategies and if you are not using mixed precision. Otherwise upgrading the version is probably the easist.

**Comment by EricKani** [comment]
> 

Thanks a lot for your quick response!

**Comment by EricKani** [comment]
> Sure, it will work for most strategies and if you are not using mixed precision. Otherwise upgrading the version is probably the easist.

And I find there are some mismatching between code and documents. 
For fabric.clip_gradients(), in document, it says that the optimizer argument is optional. when I don't fill this parameter to the clip_gradients(), the terminal gives me "clip_gradients() missing 1 required positional argument: 'optimizer' ".

I have checked the version of lightning are all 2.0.2.

Thanks~
Eric

**Comment by awaelchli** [comment]
It is required to pass in the optimizer. Addressing this in the docs in #17470

**Comment by EricKani** [comment]
Thanks!

**Comment by richardrl** [comment]
@awaelchli If the model was setup with:
        self.model, self.optimizer = self.fabric.setup(self.model, self.optimizer)

I.e. the model is not a LightningModule, this does not work. I.e. you cannot access fabric inside the model if its not a lightningmodule
